src/main/java/ch/njol/skript/variables/TypeHints.java

```diff
@@ -36,13 +39,13 @@
  * </ul>
  */
 public class TypeHints {
-	
+
 	private static final Deque<Map<String, Class<?>>> typeHints = new ArrayDeque<>();
```

Comment:
```suggestion
	private static final Queue<Map<String, Class<?>>> TYPE_HINTS = new ArrayDeque<>();
```

Reply:
Making it thread safe could be good too, could just make it one instance per `ParserInstance`

Reply:
Could potentially store type hints directly on each Script instance too.
IIRC ParserInstance has its own data storing that may be used.

Reply:
It's related to the structure as a local variable connects to an event.